### PR TITLE
Migrate FWCore/Utilities tests to use catch2

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Changed just the tests which were not already using a testing framework.

#### PR validation:

Code compiles and new tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Utilities/test/test_catch2_value_ptr.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    edm::value_ptr<simple> e(std::move(c));
+    REQUIRE(c.get() == nullptr);
+    REQUIRE(*d == *e);
+    REQUIRE(e.operator->() == pc);
+
+    edm::value_ptr<simple> f;
+    if (f) {
+      REQUIRE(0);
+    } else {
+    }
```
Comment (makortel) [inline FWCore/Utilities/test/test_catch2_value_ptr.cc:69]: I wonder if the `else` branch is necessary

File: FWCore/Utilities/test/test_catch2_value_ptr.cc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    f = std::move(d);
+    REQUIRE(d.get() == nullptr);
+    REQUIRE(*e == *f);
+    REQUIRE(f.operator->() == pd);
+    if (f) {
+    } else {
```
Comment (makortel) [inline FWCore/Utilities/test/test_catch2_value_ptr.cc:75]: I wonder if the `if` branch is necessary

**Review by makortel — COMMENTED** [review]
Few minor comments

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Utilities/test/test_catch2_Digest.cc [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+namespace {
+  void testGivenString(std::string const& s) {
+    Digest dig1(s);
+    MD5Result r1 = dig1.digest();
+
+    Digest dig2;
+    dig2.append(s);
+    MD5Result r2 = dig2.digest();
+    assert(r1 == r2);
```
[the remark this answers is not in the document]
Comment (Dr15Jones) [reply FWCore/Utilities/test/test_catch2_Digest.cc]: done

File: FWCore/Utilities/test/test_catch2_InputTag.cc [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+#include <string>
+
+namespace edm {
+  class ProductRegistry;
+}
+
+// This is just for the test. Do not dereference the pointers.
+// They points to nothing legal.
+edm::ProductRegistry* reg1 = reinterpret_cast<edm::ProductRegistry*>(1);
+edm::ProductRegistry* reg2 = reinterpret_cast<edm::ProductRegistry*>(2);
```
[the remark this answers is not in the document]
Comment (Dr15Jones) [reply FWCore/Utilities/test/test_catch2_InputTag.cc]: done

File: FWCore/Utilities/test/test_catch2_value_ptr.cc (hunk 2 above)
Comment (Dr15Jones) [reply FWCore/Utilities/test/test_catch2_value_ptr.cc]: done

File: FWCore/Utilities/test/test_catch2_value_ptr.cc (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/Utilities/test/test_catch2_value_ptr.cc]: done

**Comment by makortel** [comment]
Comparison differences are in 11634.7 (#39803)

**Comment by makortel** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**